### ISSUE TITLE
Wrangler login not working when logged out of dashboard

### ISSUE THREAD

**Issue by jkup** [cloudflare/wrangler#2043]
#### Describe the bug

When I'm logged out of cloudflare.com and try to `wrangler login` it opens the page and allows me to log in but instead of taking me to the grant permissions page it just takes me to my home page. If I open the link again manually by clicking in the terminal it takes me to the authorize page but when I click Authorize I get an error (image below) about too many API tokens have been created. If I ctrl+c wrangler and then, now that I'm logged into cloudflare.com run `wrangler login` again I get the authorization page and clicking authorize works as desired!

I think the two issues are:

1. It shouldn't take me to my home page after logging in, it should take me to the authorize screen.
2. Something is throwing a too many API tokens error when it shouldn't be.

#### Reproduce the bug

A list of steps to reproduce the bug.

1. Log out on cloudflare.com
2. mkdir test-case
3. cd test-case
4. `wrangler init --site`
5. `wrangler login`

#### Expected behavior

After logging in I should be taken to the authorize page and clicking it the first time should work without throwing an error.

#### Environment and versions

Fill out the following information about your environment.

* operating system: MacOS 11.2.3
* output of `wrangler -V`: wrangler 1.19.0
* output of `node -v`: v14.16.1
* content of `wrangler.toml`:

```
type = "webpack"
route = ''
zone_id = ''
usage_model = ''
compatibility_flags = []
workers_dev = true
site = {bucket = "",entry-point = "workers-site"}
```
![Screen Shot 2021-08-26 at 3 19 35 PM](https://user-images.githubusercontent.com/490294/131023741-99b15c0f-ad9d-46a8-a552-efda2e53a70f.png)

**Comment by nilslice** [comment]
@jkup - thanks for writing this up. 

> It shouldn't take me to my home page after logging in, it should take me to the authorize screen.

This has been a bit of an issue for a while, and I think it's due to lack of support for generalized deep linking from the dashboard.

> Something is throwing a too many API tokens error when it shouldn't be.

This seems weird, are you sure you don't have too many tokens? Maybe try on a new account and see if you can reproduce - I haven't ever seen this happen. 

---

All of this is probably less relevant, if at all, after we merge #2048 and make it the de facto login flow for `wrangler`.

**Comment by threepointone** [comment]
We landed and released #2048, so this should be resolved. Could you verify whenever you can and let us know @jkup?

**Comment by threepointone** [comment]
Closing this because I can't reproduce this anymore, but feel free to reopen if it happens again.
